torch/_dynamo/symbolic_convert.py

```diff
@@ -125,6 +126,41 @@ def generic_jump(truth_fn: typing.Callable, push: bool):
     def inner(self: "InstructionTranslatorBase", inst: Instruction):
         value: VariableTracker = self.pop()
         self.output.guards.update(value.guards)
+        if config.rewrite_assert_with_torch_assert:
+            # detect if this jump instruction is assert
+            # Python assertion is in following format:
+            # 18 POP_JUMP_IF_TRUE       24
+            # 20 LOAD_ASSERTION_ERROR
+            # 22 RAISE_VARARGS           1
+            #
+            # We find this sequence and rewrite them as torch._assert
+
+            # DETECT POP_JUMP_IF_TRUE
+            if truth_fn is operator.truth and not push:
+                current_instruction_pointer = self.instruction_pointer
+                inst = self.instructions[current_instruction_pointer]
+                # Detect LOAD_ASSERTION_ERROR
+                if inst.opname == "LOAD_ASSERTION_ERROR":
+                    current_instruction_pointer += 1
+                    if current_instruction_pointer < len(self.instructions):
+                        inst_for_raise = self.instructions[current_instruction_pointer]
+                        # DETECT RAISE_VARARGS
+                        if inst_for_raise.opname == "RAISE_VARARGS":
+                            assert_error_msg = ConstantVariable(value="assertion failure")
```

Comment:
This is not sound - the "rhs" logic of the comma for an assert can have anything in it - including logic and function calls. You need to either:

1) Trace it
2) Disallow asserts with anything except strings in that field

```
def f(x): 
    try:
        print("X is", x)
        assert(x < 3), f(1)
        return x
    except:
        pass

f(3)
```

Reply:
Sounds good. I think string is good enough for us now.